core/desugarDatatypes.ml

```diff
@@ -236,11 +306,16 @@ module Desugar = struct
     | Row r      -> Row, row alias_env r node
     | Presence f -> Presence, fieldspec alias_env f node
 
-
-
   let datatype' alias_env ((dt, _) : datatype') =
     (dt, Some (datatype alias_env dt))
 
+  let row' alias_env ((r, _) :row') =
+    (r, Some (row alias_env r (WithPos.make (Datatype.Effect r)))) (* TODO(rj) should keep the pos *)
```

Comment:
I think this is fine for now. Ideally, `row` would be part of the datatype definition -- we should work towards that at a later point.